File: wwwroot/dart/asset.dart
library asset;

class Asset {
  String _mesh;
  String _mtl;
  String _obj;
  String _shader;
  String _texture;

  String get mesh => _mesh;
  String get mtl => _mtl;
  String get obj => _obj;
  String get shader => _shader;
  String get texture => _texture;

  void setRoot(String root) {
    _mesh = "${root}mesh";
    _mtl = "${root}mtl";
    _obj = "${root}obj";
    _shader = "${root}shader";
    _texture = "${root}texture";
  }

  Asset(String root) {
    setRoot(root);
  }
}

File: webserv/src/negentropia/negoc/asset.go
package main

import (
	"fmt"
	//"negentropia/world/parser"
	//"strings"
)

type asset struct {
	mesh    string
	mtl     string
	obj     string
	shader  string
	texture string
}

func (a *asset) setRoot(path string) {
	log(fmt.Sprintf("asset.setRoot: setting asset root to path=%s", path))

	a.mesh = fmt.Sprintf("%smesh", path)
	a.mtl = fmt.Sprintf("%smtl", path)
	a.obj = fmt.Sprintf("%sobj", path)
	a.shader = fmt.Sprintf("%sshader", path)
	a.texture = fmt.Sprintf("%stexture", path)
}
